src/net/data_stream.rs

```diff
@@ -73,216 +107,143 @@ pub struct AdcDacData {
 /// `stream` is the logically consumer (UDP transmitter) of the enqueued data.
 pub fn setup_streaming(
     stack: NetworkReference,
-) -> (BlockGenerator, DataStream) {
-    let queue = cortex_m::singleton!(: Queue<AdcDacData, BLOCK_BUFFER_SIZE> = Queue::new()).unwrap();
-
+) -> (FrameGenerator, DataStream) {
+    let queue =
+        cortex_m::singleton!(: Queue<StreamFrame, FRAME_COUNT> = Queue::new())
```

Comment:
The `Queue` items should not be more than `Box<[u8; FRAME_SIZE]>`. I'd make `setup_streaming` generic over batch size, batches_per_frame and add `format` to `FrameGenerator::new()`. That also allows making the frame an exact multiple of the batch size plus header size.

Reply:
Yeah, I had originally implemented it similar to this, but I didn't like how the generic parameters started propagating through a lot of the code base. In any case, it still seems to make more sense to make it a const-generic parameter. Will update.

Reply:
Maybe with a few associated types as shorthands?

Reply:
I'm electing to leave this without all the generic parameters - it started getting pretty out-of-hand and didn't support use cases where streaming wasn't desired at all. The `Queue` is now just the box'd array + a length, so it's not much overhead.

Reply:
Why length if the frame is always full?